src/tools/imgtool/charconv.h

```diff
@@ -9,23 +9,72 @@
 ***************************************************************************/
 
 
-#ifndef __CHARCONV_H__
-#define __CHARCONV_H__
+#ifndef IMGTOOL_CHARCONV_H
+#define IMGTOOL_CHARCONV_H
 
+#include "unicode.h"
 
-/* Supported character sets */
-enum imgtool_charset
+namespace imgtool
 {
-	IMGTOOL_CHARSET_UTF8,
-	IMGTOOL_CHARSET_ISO_8859_1,
-};
+	// ======================> charconverter
+
+	// abstract base class for character conversions
+	class charconverter
+	{
+	public:
+		virtual void from_utf8(std::ostream &dest, const std::string &src) const = 0;
+		virtual void to_utf8(std::ostream &dest, const std::string &src) const = 0;
```

Comment:
Why require the input to be in the form of a `std::string`?  Why not just pointer and size for a contiguous sequence of bytes?  Text may not be stored in a string, and you might not always want to convert an entire string.